vllm/scripts.py

```diff
@@ -0,0 +1,193 @@
+# The CLI entrypoint to vLLM.
+import argparse
+import requests
+from typing import Dict, Any
+
+from vllm.entrypoints.openai.api_server import run_server
+from vllm.entrypoints.openai.cli_args import make_arg_parser
+from vllm.entrypoints.openai.protocol import (CompletionRequest, ChatCompletionRequest)
+
+COMPLETE_ROUTE = "/v1/completions"
+CHAT_COMPLETE_ROUTE = "/v1/chat/completions"
+LIST_MODELS_ROUTE = "/v1/models"
+
+def main():
+    parser = argparse.ArgumentParser(description="vLLM CLI")
+    subparsers = parser.add_subparsers(required=True)
+
+    serve_parser = subparsers.add_parser(
+        "serve",
+        help="Start the vLLM OpenAI Compatible API server",
+        usage="vllm serve <model_tag> [options]")
+    # Override the `--model` optional argument, make it positional.
+    serve_parser.add_argument(
+        "model-tag",
+        type=str,
+        help="The model tag to serve")
+    serve_parser = make_arg_parser(serve_parser)
+    serve_parser.set_defaults(func=run_server)
+    
+    complete_parser = subparsers.add_parser(
+        "complete",
+        help="Generate text completions based on the given prompt via the running API server",
+        usage="vllm complete [options]")
+    complete_parser.add_argument("--url",
+                                 type=str,
+                                 default=None,
```

Comment:
default to "http://localhost:8000" so user can put in `https://api.openai.com` as well

Reply:
Yes let's also add an option for api-key